py/src/braintrust/logger.py

```diff
@@ -370,9 +371,10 @@ def log(self, *args):
         self.queue_filled_semaphore.release()
 
     def _start(self):
-        if not self.started:
-            self.thread.start()
-            self.started = True
+        with self.start_thread_lock:
+            if not self.started:
+                self.thread.start()
+                self.started = True
```

Comment:
This is going to add contention in the common case too. You can tweak this slightly to:

```
    if not self.started:
        with self.start_thread_lock:
            if not self.started:
                self.thread.start()
                self.started = True
```